spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TagExtension.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spockframework.runtime.extension.builtin;
+
+import org.spockframework.runtime.extension.IAnnotationDrivenExtension;
+import org.spockframework.runtime.model.FeatureInfo;
+import org.spockframework.runtime.model.SpecElementInfo;
+import org.spockframework.runtime.model.SpecInfo;
+import spock.lang.Tag;
+
+/**
+ * Processes {@link Tag @Tag} annotations on specifications and features, converting them to
+ * {@link org.spockframework.runtime.model.Tag Spock tags} which can potentially be used by reporting extensions.
+ * More importantly, it also makes the tag information available to JUnit 5 as
+ * {@link org.junit.platform.engine.TestTag platform tags} in order to enable filtering, e.g. from build tools like
+ * Maven or Gradle.
+ *
+ * @see <a href="https://junit.org/junit5/docs/current/user-guide/#running-tests-tag-expressions">
+ *   JUnit platform tag expression syntax</a>
+ * @author Alexander Kriegisch
+ * @since 2.2
+ */
+public class TagExtension implements IAnnotationDrivenExtension<Tag> {
```

Comment:
❔  Currently, it seems that you can't override `@Tag` aren't there scenarios, where you might want to replace tags instead of merging them?

Reply:
Of course we could add more features or flexibility to Spock's JUnit tag logic than the Jupiter one has, being in no way obliged to implement it like Jupiter does it or to even implement it via annotations. But AFAIK, there is is always accumulative. There might be use cases in which a user might want to say: "Use feature tags only here, disregard spec tags." Or: "Ignore parent spec tags here." In order to achieve that, we could add an optional boolean parameter to the annotation, which then we would have to process when creating the platform tags. But actually, I would not want to let features creep into the product which yet no user has asked for, but we would have to cover with tests and maintain. This case is different from the two ways to apply multiple tags we are discussing above, because there is precedence, here we are on a green field.

Which case do you have in mind? Something like "The class is category 'integration test', but this feature is not" does not make much sense. Maybe something like "The spec is generally slow, but this one method is fast". Hm, I don't know if this is really a use case we need to support right now. WDYT?